src/index.js

```diff
@@ -20,6 +20,200 @@ import {
+      request = { method: 'eth_newBlockFilter', params: [] };
+    } else if (filterType === 'pendingTransaction') {
+      request = { method: 'eth_newPendingTransactionFilter', params: [] };
+    }
+
+    filterId = await ethereum.request(request);
+
+    pollingTimer = setInterval(async () => {
+      const filterChanges = await ethereum.request({
+        method: 'eth_getFilterChanges',
+        params: [filterId],
+      });
+
+      console.log(`${filterType} filter changes`, filterChanges);
+
+      if (filterChanges.length > 0) {
+        const logMessage = `[${new Date().toISOString()}] ${JSON.stringify(
+          filterChanges,
+          null,
+          '  ',
+        )}`;
+        if (filterChangesBox.innerHTML === filterChangesBoxPlaceholder) {
+          filterChangesBox.innerHTML = logMessage;
+        } else {
+          filterChangesBox.innerHTML = `${logMessage}\n\n${filterChangesBox.innerHTML}`;
+        }
+      }
+    }, 2000);
+
+    createFilterButton.disabled = true;
+    createFilterButton.innerHTML = 'Filter created, waiting for blocks...';
+    removeFilterButton.disabled = false;
+  }
+
+  async function removeFilter() {
+    await ethereum.request({
+      method: 'eth_uninstallFilter',
+      params: [filterId],
+    });
+    clearInterval(pollingTimer);
+    reset({ includingFilterChangesBox: false });
+  }
+
+  function reset({ includingFilterChangesBox = true } = {}) {
+    removeFilterButton.disabled = true;
+    createFilterButton.disabled = false;
+    createFilterButton.innerHTML = `Create ${filterType} filter`;
+
+    if (includingFilterChangesBox) {
+      filterChangesBox.innerHTML = filterChangesBoxPlaceholder;
+    }
+  }
+}
+
+function setupSubscriptions(subscriptionType) {
+  const capitalizedSubscriptionType =
+    subscriptionType[0].toUpperCase() + subscriptionType.slice(1);
+  let newSubscriptionId;
+  const startSubscriptionButton = findElementById(
+    `start${capitalizedSubscriptionType}SubscriptionButton`,
+  );
+  const stopSubscriptionButton = findElementById(
+    `stop${capitalizedSubscriptionType}SubscriptionButton`,
+  );
+  const resultsBox = findElementById(
+    `${subscriptionType}SubscriptionResultsBox`,
+  );
+  const RESULTS_BOX_PLACEHOLDER =
+    '(Information on new blocks will appear here\nas they are created.)';
+
+  return {
+    startSubscriptionButton,
+    stopSubscriptionButton,
+    activateSubscriptionButtons,
+    reset,
+  };
+
+  function activateSubscriptionButtons() {
+    startSubscriptionButton.addEventListener('click', () => {
+      startSubscription().catch(console.error);
+    });
+
+    stopSubscriptionButton.addEventListener('click', () => {
+      stopSubscription().catch(console.error);
+    });
+  }
+
+  async function startSubscription() {
+    window.ethereum.addListener('message', onMessageReceived);
+
+    newSubscriptionId = await ethereum.request({
+      method: 'eth_subscribe',
+      params: [subscriptionType],
```

Comment:
For the `logs`  subscription type, this ends up being _very_ noisy. Perhaps we could add a "topic" filter to reduce the volume?